<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: N-integration with Projection to Latent Structures models...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for block.plsda {mixOmics}"><tr><td>block.plsda {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>N-integration with Projection to Latent Structures models (PLS) with Discriminant Analysis</h2>

<h3>Description</h3>

<p>Integration of multiple data sets measured on the same samples or observations to classify a discrete outcome, ie. N-integration with Discriminant Analysis. The method is partly based on Generalised Canonical Correlation Analysis.</p>


<h3>Usage</h3>

<pre>
block.plsda(X,
Y,
indY,
ncomp = 2,
design,
scheme,
mode,
scale = TRUE,
init = "svd",
tol = 1e-06,
max.iter = 100,
near.zero.var = FALSE,
all.outputs = TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>A list of data sets (called 'blocks') measured on the same samples. Data in the list should be arranged in matrices, samples x variables, with samples order matching in all data sets.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>A factor or a class vector indicating the discrete outcome of each sample.</p>
</td></tr>
<tr valign="top"><td><code>indY</code></td>
<td>
<p>To be supplied if Y is missing, indicates the position of the factor / class vector outcome in the list <code>X</code></p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components to include in the model. Default to 2. Applies to all blocks.</p>
</td></tr>
<tr valign="top"><td><code>design</code></td>
<td>
<p>numeric matrix of size (number of blocks in X) x (number of blocks in X) with values between 0 and 1. Each value indicates the strenght of the relationship to be modelled between two blocks; a value of 0 indicates no relationship, 1 is the maximum value. If <code>Y</code> is provided instead of <code>indY</code>, the <code>design</code> matrix is changed to include relationships to <code>Y</code>. </p>
</td></tr>
<tr valign="top"><td><code>scheme</code></td>
<td>
<p>Either &quot;horst&quot;, &quot;factorial&quot; or &quot;centroid&quot;. Default = <code>horst</code>, see reference.</p>
</td></tr>
<tr valign="top"><td><code>mode</code></td>
<td>
<p>character string. What type of algorithm to use, (partially) matching
one of <code>"regression"</code>, <code>"canonical"</code>, <code>"invariant"</code> or <code>"classic"</code>.
See Details. Default = <code>regression</code>.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>boleean. If scale = TRUE, each block is standardized
to zero means and unit variances. Default = <code>TRUE</code>.</p>
</td></tr>
<tr valign="top"><td><code>init</code></td>
<td>
<p>Mode of initialization use in the algorithm, either by Singular Value Decompostion of the product of each block of X with Y (&quot;svd&quot;) or each block independently (&quot;svd.single&quot;). Default = <code>svd</code>.</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>Convergence stopping value.</p>
</td></tr>
<tr valign="top"><td><code>max.iter</code></td>
<td>
<p>integer, the maximum number of iterations.</p>
</td></tr>
<tr valign="top"><td><code>near.zero.var</code></td>
<td>
<p>boolean, see the internal <code><a href="nearZeroVar.html">nearZeroVar</a></code> function (should be set to TRUE in particular for data with many zero values). Default = <code>FALSE</code>.</p>
</td></tr>
<tr valign="top"><td><code>all.outputs</code></td>
<td>
<p>boolean. Computation can be faster when some specific (and non-essential) outputs are not calculated. Default = <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>block.plsda</code> function fits a horizontal integration PLS-DA model with a specified number of components per block).
A factor indicating the discrete outcome needs to be provided, either by <code>Y</code> or by its position <code>indY</code> in the list of blocks <code>X</code>.
</p>
<p><code>X</code> can contain missing values. Missing values are handled by being disregarded during the cross product computations in the algorithm <code>block.pls</code> without having to delete rows with missing data. Alternatively, missing data can be imputed prior using the  <code>nipals</code> function.
</p>
<p>The type of algorithm to use is specified with the <code>mode</code> argument. Four PLS
algorithms are available: PLS regression <code>("regression")</code>, PLS canonical analysis
<code>("canonical")</code>, redundancy analysis <code>("invariant")</code> and the classical PLS
algorithm <code>("classic")</code> (see References and <code>?pls</code> for more details).
</p>
<p>Note that our method is partly based on Generalised Canonical Correlation Analysis and differs from the MB-PLS approaches proposed by Kowalski et al., 1989, J Chemom 3(1) and Westerhuis et al., 1998, J Chemom, 12(5).
</p>


<h3>Value</h3>

<p><code>block.plsda</code> returns an object of class <code>"block.plsda","block.pls"</code>, a list
that contains the following components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>the centered and standardized original predictor matrix.</p>
</td></tr>
<tr valign="top"><td><code>indY</code></td>
<td>
<p>the position of the outcome Y in the output list X.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components included in the model for each block.</p>
</td></tr>
<tr valign="top"><td><code>mode</code></td>
<td>
<p>the algorithm used to fit the model.</p>
</td></tr>
<tr valign="top"><td><code>variates</code></td>
<td>
<p>list containing the variates of each block of X.</p>
</td></tr>
<tr valign="top"><td><code>loadings</code></td>
<td>
<p>list containing the estimated loadings for the variates.</p>
</td></tr>
<tr valign="top"><td><code>names</code></td>
<td>
<p>list containing the names to be used for individuals and variables.</p>
</td></tr>
<tr valign="top"><td><code>nzv</code></td>
<td>
<p>list containing the zero- or near-zero predictors information.</p>
</td></tr>
<tr valign="top"><td><code>iter</code></td>
<td>
<p>Number of iterations of the algorthm for each component</p>
</td></tr>
<tr valign="top"><td><code>explained_variance</code></td>
<td>
<p>Percentage of explained variance for each component and each block</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Florian Rohart, Benoit Gautier, Kim-Anh Lê Cao</p>


<h3>References</h3>

<p>On PLSDA:
</p>
<p>Barker M and Rayens W (2003). Partial least squares for discrimination. <em>Journal of Chemometrics</em> <b>17</b>(3), 166-173.
Perez-Enciso, M. and Tenenhaus, M. (2003). Prediction of clinical outcome with microarray data:
a partial least squares discriminant analysis (PLS-DA) approach. <em>Human Genetics</em>
<b>112</b>, 581-592.
Nguyen, D. V. and Rocke, D. M. (2002). Tumor classification by partial
least squares using microarray gene expression data. <em>Bioinformatics</em>
<b>18</b>, 39-50.
</p>
<p>On multiple integration with PLS-DA:
Gunther O., Shin H., Ng R. T. , McMaster W. R., McManus B. M. , Keown P. A. , Tebbutt S.J. , Lê Cao K-A. ,  (2014) Novel multivariate methods for integration of genomics and proteomics data: Applications in a kidney transplant rejection study, OMICS: A journal of integrative biology, 18(11), 682-95.
</p>
<p>On multiple integration with sPLS-DA and 4 data blocks:
</p>
<p>Singh A., Gautier B., Shannon C., Vacher M., Rohart F., Tebbutt S. and Lê Cao K.A. (2016).
DIABLO:  multi omics integration for biomarker discovery. BioRxiv available here: <a href="http://biorxiv.org/content/early/2016/08/03/067611">http://biorxiv.org/content/early/2016/08/03/067611</a>
</p>
<p>mixOmics article:
</p>
<p>Rohart F, Gautier B, Singh A, Lê Cao K-A. mixOmics: an R package for 'omics feature selection and multiple data integration. PLoS Comput Biol 13(11): e1005752
</p>


<h3>See Also</h3>

<p><code><a href="plotIndiv.html">plotIndiv</a></code>, <code><a href="plotArrow.html">plotArrow</a></code>, <code><a href="plotLoadings.html">plotLoadings</a></code>, <code><a href="plotVar.html">plotVar</a></code>, <code><a href="predict.html">predict</a></code>, <code><a href="perf.html">perf</a></code>, <code><a href="selectVar.html">selectVar</a></code>, <code><a href="block.pls.html">block.pls</a></code>, <code><a href="block.splsda.html">block.splsda</a></code> and http://www.mixOmics.org for more details.</p>


<h3>Examples</h3>

<pre>

data(nutrimouse)
data = list(gene = nutrimouse$gene, lipid = nutrimouse$lipid, Y = nutrimouse$diet)
# with this design, all blocks are connected
design = matrix(c(0,1,1,1,0,1,1,1,0), ncol = 3, nrow = 3,
byrow = TRUE, dimnames = list(names(data), names(data)))

res = block.plsda(X = data, indY = 3) # indY indicates where the outcome Y is in the list X
plotIndiv(res, ind.names = FALSE, legend = TRUE)
plotVar(res)

## Not run: 
# when Y is provided
res2 = block.plsda(list(gene = nutrimouse$gene, lipid = nutrimouse$lipid),
Y = nutrimouse$diet, ncomp = 2)
plotIndiv(res2)
plotVar(res2)

## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.10.8 <a href="00Index.html">Index</a>]</div>
</body></html>
