<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: P-integration</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for mint.pls {mixOmics}"><tr><td>mint.pls {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>P-integration</h2>

<h3>Description</h3>

<p>Function to integrate and combine multiple independent studies measured on the same variables or predictors (P-integration) using variants of multi-group PLS (unsupervised analysis).
</p>


<h3>Usage</h3>

<pre>
mint.pls(X,
Y,
ncomp = 2,
mode = c("regression", "canonical", "invariant", "classic"),
study,
scale = TRUE,
tol = 1e-06,
max.iter = 100,
near.zero.var = FALSE,
all.outputs = TRUE)

</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>numeric matrix of predictors combining multiple independent studies on the same set of predictors. <code>NA</code>s are allowed.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>Matrix or vector response for a multivariate regression framework. Data should be continuous variables (see <code>mint.plsda</code> for supervised classification and factor reponse)</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>Number of components to include in the model (see Details). Default to 2</p>
</td></tr>
<tr valign="top"><td><code>mode</code></td>
<td>
<p>character string. What type of algorithm to use, (partially) matching
one of <code>"regression"</code> or <code>"canonical"</code>. See Details.</p>
</td></tr>
<tr valign="top"><td><code>study</code></td>
<td>
<p>factor indicating the membership of each sample to each of the studies being combined</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>boleean. If scale = TRUE, each block is standardized
to zero means and unit variances. Default = <code>TRUE</code>.</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>Convergence stopping value.</p>
</td></tr>
<tr valign="top"><td><code>max.iter</code></td>
<td>
<p>integer, the maximum number of iterations.</p>
</td></tr>
<tr valign="top"><td><code>near.zero.var</code></td>
<td>
<p>boolean, see the internal <code><a href="nearZeroVar.html">nearZeroVar</a></code> function (should be set to TRUE in particular for data with many zero values). Default = <code>FALSE</code>.</p>
</td></tr>
<tr valign="top"><td><code>all.outputs</code></td>
<td>
<p>boolean. Computation can be faster when some specific (and non-essential) outputs are not calculated. Default = <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mint.pls</code> fits a vertical PLS-DA models with <code>ncomp</code> components in which several independent studies measured on the same variables are integrated. The aim is to explain the continuous outcome <code>Y</code>. The <code>study</code> factor indicates the membership of each sample in each study. We advise to only combine studies with more than 3 samples as the function performs internal scaling per study.
</p>
<p>Multi (continuous)response are supported. <code>X</code> and <code>Y</code> can contain missing values. Missing values are handled by being disregarded during the cross product computations in the algorithm <code>mint.pls</code> without having to delete rows with missing data. Alternatively, missing data can be imputed prior using the  <code>nipals</code> function.
</p>
<p>The type of algorithm to use is specified with the <code>mode</code> argument. Four PLS
algorithms are available: PLS regression <code>("regression")</code>, PLS canonical analysis
<code>("canonical")</code>, redundancy analysis <code>("invariant")</code> and the classical PLS
algorithm <code>("classic")</code> (see References and more details in <code>?pls</code>).
</p>
<p>Useful graphical outputs are available, e.g. <code><a href="plotIndiv.html">plotIndiv</a></code>, <code><a href="plotLoadings.html">plotLoadings</a></code>, <code><a href="plotVar.html">plotVar</a></code>.
</p>


<h3>Value</h3>

<p><code>mint.pls</code> returns an object of class <code>"mint.pls", "pls"</code>, a list
that contains the following components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>the centered and standardized original predictor matrix.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>the centered and standardized original response vector or matrix.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components included in the model.</p>
</td></tr>
<tr valign="top"><td><code>study</code></td>
<td>
<p>The study grouping factor</p>
</td></tr>
<tr valign="top"><td><code>mode</code></td>
<td>
<p>the algorithm used to fit the model.</p>
</td></tr>
<tr valign="top"><td><code>variates</code></td>
<td>
<p>list containing the variates of X - global variates.</p>
</td></tr>
<tr valign="top"><td><code>loadings</code></td>
<td>
<p>list containing the estimated loadings for the variates - global loadings.</p>
</td></tr>
<tr valign="top"><td><code>variates.partial</code></td>
<td>
<p>list containing the variates of X relative to each study - partial variates.</p>
</td></tr>
<tr valign="top"><td><code>loadings.partial</code></td>
<td>
<p>list containing the estimated loadings for the partial variates - partial loadings.</p>
</td></tr>
<tr valign="top"><td><code>names</code></td>
<td>
<p>list containing the names to be used for individuals and variables.</p>
</td></tr>
<tr valign="top"><td><code>nzv</code></td>
<td>
<p>list containing the zero- or near-zero predictors information.</p>
</td></tr>
<tr valign="top"><td><code>iter</code></td>
<td>
<p>Number of iterations of the algorthm for each component</p>
</td></tr>
<tr valign="top"><td><code>explained_variance</code></td>
<td>
<p>Percentage of explained variance for each component and each study (note that contrary to PCA, this amount may not decrease as the aim of the method is not to maximise the variance, but the covariance between data sets).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Florian Rohart, Kim-Anh Lê Cao</p>


<h3>References</h3>

<p>Rohart F, Eslami A, Matigian, N, Bougeard S, Lê Cao K-A (2017). MINT: A multivariate integrative approach to identify a reproducible biomarker signature across multiple experiments and platforms. BMC Bioinformatics 18:128.
</p>
<p>Eslami, A., Qannari, E. M., Kohler, A., and Bougeard, S. (2014). Algorithms for multi-group PLS. J. Chemometrics, 28(3), 192-201.
</p>


<h3>See Also</h3>

<p><code><a href="spls.html">spls</a></code>, <code><a href="summary.html">summary</a></code>,
<code><a href="plotIndiv.html">plotIndiv</a></code>, <code><a href="plotVar.html">plotVar</a></code>, <code><a href="../../stats/html/predict.html">predict</a></code>, <code><a href="perf.html">perf</a></code>, <code><a href="mint.spls.html">mint.spls</a></code>, <code><a href="mint.plsda.html">mint.plsda</a></code>, <code><a href="mint.splsda.html">mint.splsda</a></code> and http://www.mixOmics.org/mixMINT for more details.</p>


<h3>Examples</h3>

<pre>
data(stemcells)

# for the purpose of this example, we artificially
# create a continuous response Y by taking gene 1.

res = mint.pls(X = stemcells$gene[,-1], Y = stemcells$gene[,1], ncomp = 3,
study = stemcells$study)

plotIndiv(res)

#plot study-specific outputs for all studies
plotIndiv(res, study = "all.partial")

## Not run: 
#plot study-specific outputs for study "2"
plotIndiv(res, study = "2", col = 1:3, legend = TRUE)

## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.6.2 <a href="00Index.html">Index</a>]</div>
</body></html>
