<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Arrow sample plot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for plotArrow {mixOmics}"><tr><td>plotArrow {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Arrow sample plot</h2>

<h3>Description</h3>

<p>Represents samples from multiple coordinates.
</p>


<h3>Usage</h3>

<pre>
plotArrow(  object,
comp = NULL,
abline = FALSE,
xlim = NULL,
ylim = NULL,
group=NULL,
col,
cex,
pch,
title=NULL,
plot.arrows=TRUE,
legend=FALSE,
X.label = NULL,
Y.label = NULL,
ind.names=FALSE,
position.names='centroid'
)

</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>object</code></td>
<td>
<p>object of class inheriting from <span class="pkg">mixOmics</span>:  <code>PLS, sPLS, rCC, rGCCA, sGCCA, sGCCDA</code></p>
</td></tr>
<tr valign="top"><td><code>comp</code></td>
<td>
<p>integer vector of length two indicating the components represented on the horizontal
and the vertical axis to project the individuals.</p>
</td></tr>
<tr valign="top"><td><code>abline</code></td>
<td>
<p>should the vertical and horizontal line through the center be plotted? Default set to <code>FALSE</code></p>
</td></tr>
<tr valign="top"><td><code>xlim</code></td>
<td>
<p>the ranges to be encompassed by the <i>x</i> axis, if <code>NULL</code> they are computed.</p>
</td></tr>
<tr valign="top"><td><code>ylim</code></td>
<td>
<p>the ranges to be encompassed by the <i>y</i> axis, if <code>NULL</code> they are computed.</p>
</td></tr>
<tr valign="top"><td><code>group</code></td>
<td>
<p>factor indicating the group membership for each sample. Coded as default for the supervised method <code>sGCCDA, sPLSDA</code>, but needs to be input for the unsupervised methods <code>PLS, sPLS, rCC, rGCCA, sGCCA</code></p>
</td></tr>
<tr valign="top"><td><code>col</code></td>
<td>
<p>character (or symbol) color to be used, color vector also possible.</p>
</td></tr>
<tr valign="top"><td><code>cex</code></td>
<td>
<p>numeric character (or symbol) expansion, , color vector also possible.</p>
</td></tr>
<tr valign="top"><td><code>pch</code></td>
<td>
<p>plot character. A character string or a vector of single characters
or integers. See <code><a href="../../graphics/html/points.html">points</a></code> for all alternatives.</p>
</td></tr>
<tr valign="top"><td><code>title</code></td>
<td>
<p>set of characters for the title plot.</p>
</td></tr>
<tr valign="top"><td><code>plot.arrows</code></td>
<td>
<p>boolean. Whether arrows should be added or not. Default is <code>TRUE</code>.</p>
</td></tr>
<tr valign="top"><td><code>legend</code></td>
<td>
<p>boolean. Whether the legend should be added. Only for the supervised methods and if group!=NULL. Default is <code>FALSE</code>.</p>
</td></tr>
<tr valign="top"><td><code>X.label</code></td>
<td>
<p>x axis titles.</p>
</td></tr>
<tr valign="top"><td><code>Y.label</code></td>
<td>
<p>y axis titles.</p>
</td></tr>
<tr valign="top"><td><code>ind.names</code></td>
<td>
<p>If <code>TRUE</code>, the row names of the first (or second) data matrix are used as sample names (see Details). Can be a vector of length the sample size to display sample names.</p>
</td></tr>
<tr valign="top"><td><code>position.names</code></td>
<td>
<p>One of <code>"centroid", "start", "end"</code>. Define where sample names are plotted when <code>ind.names=TRUE</code>. In a multiblock analysis, centroid and start will display similarly. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Graphical of the samples (individuals) is displayed in a superimposed manner where each sample will be
indicated using an arrow. The start of the arrow indicates the location of the sample in <i>X</i> in one plot,
and the tip the location of the sample in <i>Y</i> in the other plot.
</p>
<p>For objects of class <code>"GCCA"</code> and if there are more than 3 blocks, the start of the  arrow indicates the centroid between all data sets for a given individual and the tips of the arrows the location of that individual in each block.
</p>
<p>Short arrows indicate a strong agreement between the matching data sets, long arrows a disagreement between the matching data sets.
</p>


<h3>Value</h3>

<p>none</p>


<h3>Author(s)</h3>

<p>Francois Bartolo, Kim-Anh Lê Cao.</p>


<h3>References</h3>

<p>Lê Cao, K.-A., Martin, P.G.P., Robert-Granie, C. and Besse, P. (2009). Sparse canonical methods
for biological data integration: application to a cross-platform study. <em>BMC Bioinformatics</em> <b>10</b>:34.
</p>


<h3>See Also</h3>

<p><code><a href="../../graphics/html/arrows.html">arrows</a></code>, <code><a href="../../graphics/html/text.html">text</a></code>, <code><a href="../../graphics/html/points.html">points</a></code> and http://mixOmics.org/graphics for more details.</p>


<h3>Examples</h3>

<pre>
## plot of individuals for objects of class 'rcc'
# ----------------------------------------------------
dev.off()
data(nutrimouse)
X &lt;- nutrimouse$lipid
Y &lt;- nutrimouse$gene
nutri.res &lt;- rcc(X, Y, ncomp = 3, lambda1 = 0.064, lambda2 = 0.008)

plotArrow(nutri.res)

# names indicate genotype
plotArrow(nutri.res,
group = nutrimouse$genotype, ind.names = nutrimouse$genotype)


plotArrow(nutri.res, group = nutrimouse$genotype,
legend = TRUE)


## Not run: 
## plot of individuals for objects of class 'pls' or 'spls'
# ----------------------------------------------------
data(liver.toxicity)
X &lt;- liver.toxicity$gene
Y &lt;- liver.toxicity$clinic
toxicity.spls &lt;- spls(X, Y, ncomp = 3, keepX = c(50, 50, 50),
keepY = c(10, 10, 10))

#default
plotArrow(toxicity.spls)


# colors indicate time of necropsy, text is the dose
plotArrow(toxicity.spls,  group = liver.toxicity$treatment[, 'Time.Group'],
ind.names  = liver.toxicity$treatment[, 'Dose.Group'],
legend = TRUE)

# colors indicate time of necropsy, text is the dose, label at start of arrow
plotArrow(toxicity.spls,  group = liver.toxicity$treatment[, 'Time.Group'],
ind.names  = liver.toxicity$treatment[, 'Dose.Group'],
legend = TRUE, position.names = 'start')

## variable representation for objects of class 'sgcca' (or 'rgcca')
# ----------------------------------------------------
data(nutrimouse)
Y = unmap(nutrimouse$diet)
data = list(gene = nutrimouse$gene, lipid = nutrimouse$lipid, Y = Y)
design1 = matrix(c(0,1,1,1,0,1,1,1,0), ncol = 3, nrow = 3, byrow = TRUE)
nutrimouse.sgcca &lt;- wrapper.sgcca(X = data,
design = design1,
penalty = c(0.3, 0.5, 1),
ncomp = 3,
scheme = "centroid")

# default style: same color for all samples
plotArrow(nutrimouse.sgcca)


plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot')

# ind.names to visualise the unique individuals
plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot', ind.names = TRUE)

# ind.names to visualise the unique individuals
plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot', ind.names = TRUE,position.names   = 'start')

plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot', ind.names = TRUE,position.names   = 'end')

# ind.names indicates the diet
plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot', ind.names = nutrimouse$diet, position.names= 'start')

# ind.names to visualise the unique individuals, start position
plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot', ind.names = TRUE, position.names   = 'start')

# end position
plotArrow(nutrimouse.sgcca, group = nutrimouse$diet, legend =TRUE,
title = 'my plot', ind.names = TRUE, position.names   = 'end')


## variable representation for objects of class 'sgccda'
# ----------------------------------------------------
# Note: the code differs from above as we use a 'supervised' GCCA analysis
data(nutrimouse)
Y = nutrimouse$diet
data = list(gene = nutrimouse$gene, lipid = nutrimouse$lipid)
design1 = matrix(c(0,1,0,1), ncol = 2, nrow = 2, byrow = TRUE)

nutrimouse.sgccda1 &lt;- wrapper.sgccda(X = data,
Y = Y,
design = design1,
ncomp = 2,
keepX = list(gene = c(10,10), lipid = c(15,15)),
scheme = "centroid")


#  default colors correspond to outcome Y
plotArrow(nutrimouse.sgccda1)


# with legend and title and indiv ID
plotArrow(nutrimouse.sgccda1,  legend = TRUE, title = 'my sample plot',
ind.names = TRUE, position.names = 'start')


## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.6.2 <a href="00Index.html">Index</a>]</div>
</body></html>
